src/lib/vasi-macro/src/lib.rs

```diff
@@ -259,6 +284,10 @@ fn impl_derive_virtual_address_space_independent(ast: syn::DeriveInput) -> proc_
                 #calls_to_check
             };
         }
+        #[deny(improper_ctypes_definitions)]
+        const _: () = {
+            extern "C" fn _vasi_validate_ffi_safe #impl_generics (_: #name #ty_generics) #where_clause {}
+        };
```

Comment:
maybe add a comment about why this exists?

Reply:
done